client/app/pods/components/type-ahead/component.coffee

```diff
@@ -0,0 +1,54 @@
+`import Ember from 'ember'`
+
+TypeAheadComponent = Ember.TextField.extend
+
+  suggestionEngine: null
+  displayKey: null
+  selection: null
+  url: null
+
+  setupSuggestionEngine:(->
+    self = @
+    @_super()
+    @suggestionEngine = new Bloodhound(
```

Comment:
I suggest waiting to do this until `didInsertElement`, because that will speed up the initial render of the app.